core/src/main/scala/tamer/Tamer.scala

```diff
@@ -85,67 +77,73 @@ object Tamer {
     val key          = StateKey(stateHash.toHexString, stateGroupId)
     val subscription = Subscription.topics(stateTopic)
 
-    val waitForAssignment = consumer.assignment
-      .withFilter(_.nonEmpty)
-      .tapError(_ => log.debug(s"still no assignment on $stateGroupId, there are no partitions to process"))
-      .retry(tenTimes)
-
-    val subscribeAndDecide = log.debug(s"subscribing to $subscription") *>
-      consumer.subscribe(subscription) *>
-      log.debug("awaiting assignment") *>
-      waitForAssignment.flatMap(partitionSet =>
-        log.debug(s"received assignment $partitionSet") *>
-          consumer.endOffsets(partitionSet).map(decidedAction).tap(d => log.debug(s"decided to $d"))
-      )
+    val partitionInfo: Task[Set[TopicPartition]] = for {
+      topics        <- consumer.listTopics()
+      partitionInfo <- ZIO.fromOption(topics.get(stateTopic)).mapError(_ => TamerError(s"Group $stateGroupId has no permission on topic $stateTopic"))
+    } yield partitionInfo.map(pi => new TopicPartition(pi.topic(), pi.partition())).toSet
+
+    val stateTopicDecision = for {
+      _            <- log.debug(s"obtaining information on $stateTopic")
+      partitionSet <- partitionInfo
+      _            <- log.debug(s"received the following information on $stateTopic: $partitionSet")
+      decision     <- consumer.committed(partitionSet).map(decidedAction)
```

Comment:
if we ever committed an offset in the partitionSet then we resume

Reply:
this is a clever way of fixing the issue we were having where Tamer would not initialise if a topic was empty but offsets were not zero (as in the state topic was emptied out by delete cleanup policy)

Reply:
If only I had thought about it this way yeah 😄